### PR TITLE
ppm: Have verify_finish() consume the output of verify_start()

### PR DESCRIPTION
**Pull request by cjpatton**
This aligns better with the spec and also simplifies the code a bit.

### CODE REVIEW

File: src/vdaf.rs [hunk 1]
```diff
@@ -405,9 +406,15 @@ where
...
+        return Err(PpmError::Ppm(
+            "verify_finish(): expected at least one inbound messages; got none".to_string(),
+        ));
+    }
+
     // Combine the verifier messages.
-    let mut verifier_data = state.verifier_share.data;
-    let mut joint_rand_seed_check = vec![0; Aes128Ctr::STREAM_CIPHER_SEED_SIZE];
+    let mut joint_rand_seed = state.joint_rand_seed;
+    let mut verifier_data = vec![F::zero(); msgs[0].verifier_share.as_slice().len()];
```
Comment (tgeoghegan) [inline src/vdaf.rs:417]: meganit: it's a little odd that we use `verifier_share.as_slice()` here, but `verfier_share.data` below, in the iterator. Both achieve the same thing, but I feel like we should either make `Verifier`'s `data` field public, or have a public method allowing access to it, but not both.

File: src/vdaf.rs [hunk 2]
```diff
@@ -476,18 +480,8 @@ mod tests {
...
-                Vec::with_capacity(num_shares - 1);
-            for (j, verifier) in verifiers.iter().enumerate() {
-                if i != j {
-                    verifiers_for_aggregator.push(verifier.clone());
-                }
-            }
-
-            // Run aggregator i.
-            let output_share = verify_finish(state, verifiers_for_aggregator).unwrap();
+        for state in states.into_iter() {
```
Comment (tgeoghegan) [inline src/vdaf.rs:483]: nit
```suggestion
        for state in states {
```

**Review by tgeoghegan — CHANGES_REQUESTED** [review]

File: src/vdaf.rs [hunk 3]
```diff
@@ -41,6 +41,7 @@ use aes::Aes128Ctr;
 use getrandom::getrandom;
```
[the remark this answers is not in the document]
Comment (cjpatton) [reply src/vdaf.rs:41]: Yup, done.

File: src/vdaf.rs (hunk 1 above)
Comment (cjpatton) [reply src/vdaf.rs:417]: Good call. Replaced `verifier_share.data` below with `verifier_share.as_slice()` and made `data` private.

File: src/vdaf.rs [hunk 4]
```diff
@@ -1,7 +1,7 @@
 // SPDX-License-Identifier: MPL-2.0
 
-//! This module implements core functionality of the PPM protocol for Prio. It consists of the
-//! following functions:
+//! This module constructs  a Verifiable Distributed Aggregation Function (VDAF) from a
+//! [`pcp::Value``]. It consists of the following functions:
```
Comment (tgeoghegan) [inline src/vdaf.rs:4]:
```suggestion
//! [`pcp::Value`]. It consists of the following functions:
```

**Review by tgeoghegan — APPROVED** [review]
LGTM!

**Comment by cjpatton** [comment]
Squashed.

**Pull request merged**